src/Adapter/SearchEngine/CommandHandler/BulkDeleteSearchEngineHandler.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\SearchEngine\CommandHandler;
+
+use PrestaShop\PrestaShop\Adapter\SearchEngine\AbstractSearchEngineHandler;
+use PrestaShop\PrestaShop\Core\Domain\SearchEngine\Command\BulkDeleteSearchEngineCommand;
+use PrestaShop\PrestaShop\Core\Domain\SearchEngine\CommandHandler\BulkDeleteSearchEngineHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\SearchEngine\Exception\DeleteSearchEngineException;
+use PrestaShop\PrestaShop\Core\Domain\SearchEngine\Exception\SearchEngineException;
+use PrestaShopException;
+
+/**
+ * Handles command that deletes Search Engines in bulk action.
+ */
+final class BulkDeleteSearchEngineHandler extends AbstractSearchEngineHandler implements BulkDeleteSearchEngineHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @throws SearchEngineException
+     */
+    public function handle(BulkDeleteSearchEngineCommand $command): void
+    {
```

Comment:
sorry to be picky, but;
we want to get rid of inheritence as much as possible, so command handler should extend AbstractHandler, but use services instead.
In this case you should create a repository for SearchEngine entity in adapter which extends AbstractObjectModelRepository (you can see some some examples in currently migrating Product domain like this one https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/Product/Repository/ProductRepository.php#L336).


So in the end your handler will only call a repository bulkDelete method and provide array of Ids.

Reply:
To give some background to @zuk3975 : we try to follow https://en.wikipedia.org/wiki/Composition_over_inheritance 😉

Reply:
@zuk3975 Is this currently really required? Product is the only entity that is using the repository for command handlers right now and also this is not documented (and because of this using repository is really confusing for me). According to the discussion in Slack, using repositories is still in the early experimenting state.

Reply:
well, why wouldn't you want to reduce some code duplication and complexity by reusing some already introduced classes which was made just for that? Anyway it is not up to me to decide if it is required. Its up to - @jolelievre, @matks (this is related for most of my feedback in this PR)

Reply:
My comment is still relevant. Im not the one to decide though. :arrow_up: